src/FontManager.cpp

```diff
@@ -61,12 +61,38 @@ void FontManager::loadFonts(const QString& folder)
     QDir dir = folder;
     dir.setNameFilters(filters);
 
-    foreach (QString fontfile, dir.entryList(QDir::Files | QDir::Readable | QDir::NoDotAndDotDot)) {
-        QString fontFilePathName = QStringLiteral("%1/%2").arg(dir.absolutePath(), fontfile);
-        if (QFontDatabase::addApplicationFont(fontFilePathName) == -1) {
-            // At the point that addFonts() is called we have a GUI application
-            // in the making and can use qDebug() and not rely on iostream class
-            qWarning() << "FontManager::loadFonts() ERROR - Could not load the font in the file: " << fontFilePathName;
-        }
+    foreach (QString fontFile, dir.entryList(QDir::Files | QDir::Readable | QDir::NoDotAndDotDot)) {
+        QString fontFilePathName = QStringLiteral("%1/%2").arg(dir.absolutePath(), fontFile);
+        loadFont(fontFilePathName);
+    }
+}
+
+void FontManager::loadFont(const QString& filePath) {
+    if (fontAlreadyLoaded(filePath)) {
+        return;
     }
+
+    auto fontID = QFontDatabase::addApplicationFont(filePath);
+
+    // remember even if the font failed to load so we don't spam messages on fonts that repeat
+    rememberFont(filePath, fontID);
+
+    if (fontID == -1) {
+        qWarning() << "FontManager::loadFonts() warning - Could not load the font in the file: " << filePath;
```

Comment:
As we may be talking about font *collection* files, font may be fonts (plural) here...